cpp/modmesh/universe/polygon.hpp

```diff
@@ -1140,12 +1212,8 @@ class PolygonPad
     SimpleCollector<ssize_type> m_ends;
     std::unique_ptr<rtree_type> m_rtree;
 
-    // TODO: Add trapezoidal map data structure for polygon boolean operations
-    // Reference: http://www0.cs.ucl.ac.uk/staff/m.slater/Teaching/CG/1997-98/Solutions/Trap/
-    // Will need:
-    //  - Trapezoid decomposition structure
-    //  - Point location query structure
-    //  - Integration with node lists for synchronization
+    /// Cache of trapezoidal decompositions, keyed by polygon ID. Computed lazily on first access.
+    std::unordered_map<size_t, std::vector<Trapezoid>> m_decompositions;
```

Comment:
Do not use hash table.  Hash table is the last resort when (1) you run out of ideas or (2) you run out of time.

A has table to vector is even worth: tons of dynamic memory allocation.  It is bound to be slow.

This is why you need `TrapezoidPad`.

Reply:
Got it.